map-scripts.js

```diff
@@ -34,11 +45,97 @@ function calcRoute(event) {
     };
     directionsService.route(request, function (result, status) {
         if (status == 'OK') {
+            console.log(result);
             directionsRenderer.setDirections(result);
+            step = Math.floor(result.routes[0].overview_path.length / CITIES_LENGTH);
+            for (i=1; i < CITIES_LENGTH; i++){
+                latlon = result.routes[0].overview_path[i*step];
+                duration = getWeatherFromLatLon(event, latlon);
+            }
+            console.log(result.routes[0].legs[0].end_location);
+            getWeatherPrep(result, result.routes[0].legs[0].end_location);
+        }
+    });
+}
+
+function getWeatherFromLatLon(event, latlon) {
+    event.preventDefault();
+    var origin = document.getElementById('origin').value;
+    var destination = latlon;
+    var departTime = document.getElementById('start-time').value;
+    if (departTime) {
+        departTime = new Date(departTime);
+    } else {
+        departTime = new Date();
+    }
+    var request = {
+        origin: origin,
+        destination: destination,
+        drivingOptions: {
+            departureTime: departTime,
+        },
+        travelMode: 'DRIVING',
+    };
+    directionsService.route(request, function (result, status) {
+        if (status == 'OK') {
+            getWeatherPrep(result, latlon);
         }
     });
 }
 
+function getPoliticalTypes(address) {
+    return address.types.includes("political");
+}
+
+// this is not working because it is not free (╯°□°）╯︵ ┻━┻
+// this could have been a better way to turn latlon objects into city names
+function getReverseGeocoding(latlon) {
+    var fullURL = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latlon.lat()},${latlon.lng()}&key=AIzaSyBIwzALxUPNbatRBj3Xi1Uhp0fFzwWNBkE`;
+    fetch(fullURL)
+    .then(function(response) {
+      return response.json();
+    }).then(function(json) {
+      console.log(json);
+      location = json.address_components.filter(getPoliticalTypes)[0];
+      return location;
+    });
+}
+
+function getWeatherPrep(routeResult, latlon) {
+    var duration_result = routeResult.routes[0].legs[0].duration.text;
```

Comment:
Is this duration the time that it takes to get to the location from the starting position?

Reply:
Yeah correct me if I'm wrong/missing something, but I wasn't able to find arrival times for specific legs of the route, so I compensated by calling the route api multiple times for each leg in order to get the durations for each leg